nozl/schema/schema.go

```diff
@@ -30,42 +35,122 @@ func ParseOpenApiV3Schema(serviceID string, specFile []byte) error {
+		}
+		if pathValue.Delete != nil {
+			AddSchemaToKVStore(serviceID, pathKey, "delete", pathValue.Delete)
 		}
 	}
 
-	for pathKey, pathValue := range doc.Paths {
+	return nil
+}
 
-		if pathValue.Get != nil {
-			schemaList[pathValue.Get.OperationID] = addSchema(pathKey, "get", pathValue.Get)
-		} else if pathValue.Post != nil {
-			schemaList[pathValue.Post.OperationID] = addSchema(pathKey, "post", pathValue.Post)
-		} else if pathValue.Put != nil {
-			schemaList[pathValue.Put.OperationID] = addSchema(pathKey, "put", pathValue.Put)
-		} else if pathValue.Patch != nil {
-			schemaList[pathValue.Patch.OperationID] = addSchema(pathKey, "patch", pathValue.Patch)
-		} else if pathValue.Delete != nil {
-			schemaList[pathValue.Delete.OperationID] = addSchema(pathKey, "delete", pathValue.Delete)
+func MakeOptionalFieldsNullable(operation *openapi3.Operation) error {
+	if operation.RequestBody != nil {
+		ReqBodySchema := operation.RequestBody.Value.Content["application/x-www-form-urlencoded"].Schema.Value
+		RequiredParams := ReqBodySchema.Required
+		ReqBodyParams := ReqBodySchema.Properties
+
+		for key, val := range ReqBodyParams {
+			if stringInSlice(key, RequiredParams) == false {
+				val.Value.Nullable = true
+			}
 		}
 	}
 
-	jsonPayload, _ := json.Marshal(schemaList)
+	return nil
+}
+
+func stringInSlice(refStr string, list []string) bool {
+	for _, currStr := range list {
+		if currStr == refStr {
+			return true
+		}
+	}
+	return false
+}
 
-	_, err = schemaKv.Put(serviceID, jsonPayload)
+func AddSchemaToKVStore(serviceID string, pathKey string, httpMethod string, operation *openapi3.Operation) {
+	schemaKv, _ := eventstream.Eventstream.RetreiveKeyValStore(shared.SchemaKV)
+	MakeOptionalFieldsNullable(operation)
+	currSchema := addSchema(pathKey, httpMethod, operation)
+	operationID := operation.OperationID
+	jsonPayload, _ := json.Marshal(currSchema)
+	schemaKv.Put(fmt.Sprintf("%s-%s", serviceID, operationID), jsonPayload)
+}
 
+func ValidateOpenAPIV3Schema(msg *eventstream.Message) error {
+	schemaValid, err := GetMsgRefSchema(msg)
 	if err != nil {
-		shared.Logger.Error("Failed to save schema in data store.")
+		shared.Logger.Error(err.Error())
 		return err
 	}
+	openapi3.SchemaErrorDetailsDisabled = true
 
-	return nil
+	msgBody := msg.Body
+	fmt.Println(schemaValid)
+	fmt.Println(msgBody)
+
+	ctx := context.Background()
+	jsonData, _ := json.Marshal(&msgBody)
+	formData := strings.NewReader(string(jsonData))
+	//baseUrl := "https://api.twilio.com"
+	//rgx, _ := regexp.Compile("{AccountSid}")
+	//ep := rgx.ReplaceAllString(schemaValid.Path, "AC9f560ea30baaaf8013e4e44284eb6768")
+	//data := url.Values{}
+
+	//data.Add("To", +923244253153)
+	//data.Add("Body", "helloworld")
+	//payload := strings.NewReader("Body=Hello%20World!&To=%2B923214930471&From=%2B19034598701&edfgsgd=sdfds")
+	//payload := strings.NewReader(data.Encode())
+
+	httpReq, err := http.NewRequest("POST", schemaValid.Path, formData)
```

Comment:
Replace hard-coded HTTP method with the one specified in the OAPI spec

Reply:
Done